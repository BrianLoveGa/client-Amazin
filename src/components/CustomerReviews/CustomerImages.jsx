import React, { Component } from 'react';
import CImagesAll from '../CustomerImageCmpnt/CustomerImagesAll';

class CustomerImages extends Component {
    
    render() { 
        return ( 
            <div>
                <CImagesAll />
            </div>
         );
    }
}
 
export default CustomerImages;