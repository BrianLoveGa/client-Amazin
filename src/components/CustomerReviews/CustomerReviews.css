/* img {
    border: 4px solid red;
} */

.review-mention {
    margin-bottom: 26px!important;
}

.button {
    padding: 10px;
    color: #111111;
    background: #D7E8EA;
}

/* .profile-avatar-img {
    border-radius: 34px;
    border: 2px solid #FFF;
}

.profile-avatar {
    padding-right: 9px;
    width: 43px;  
} */

.profile-name {
    position: relative;
    color: #111;
    unicode-bidi: isolate;
    line-height: 19px;
    font-size: 15px;
}

/* .inline {

    width: 25%;
    margin: 10px;
} */
