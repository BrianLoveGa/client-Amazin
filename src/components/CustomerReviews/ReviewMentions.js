import React, { Component } from 'react';


class ReviewMentions extends Component {


    render() { 

        return ( 
            <div>
                <h3>Read reviews that mention</h3>
                <span><button>filtered button</button></span>
                <br />
                <br />
                <span><button>top reviews</button></span>
                <div><img src=""></img></div>
                <br />
                <p><div>star rating ***</div><h3><a href="/">Review headline</a></h3></p>
                <p><div>Country & Date</div></p>
                <p><div>Verified Purchase</div></p>

                <p>submitted review form goes here</p>
            </div>
         );
    }
}
 
export default ReviewMentions;