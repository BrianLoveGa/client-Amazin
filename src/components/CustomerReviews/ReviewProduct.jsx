import React, { Component } from 'react';

class ReviewProduct extends Component {
    
    render() { 
        return ( 
            <div>
                <h3>Review this product</h3>
                <h5>Share your thoughts with other customers</h5>
                <a href="link to review form"><button>Write a customer review</button></a>
            </div>
         );
    }
}
 
export default ReviewProduct;