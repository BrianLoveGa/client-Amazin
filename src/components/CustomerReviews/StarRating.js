import React, { Component } from 'react';

class StarRating extends Component {
    constructor() {
        super();
     
        this.state = {
          rating: 1
        };
    }
     
    render() { 
        return ( 
            <div>
                <h3>Star Ratings</h3>
                <br />
                <br />
            </div>
         );
    }
}
 
export default StarRating;