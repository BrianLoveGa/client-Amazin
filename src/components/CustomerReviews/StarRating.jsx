import React, { Component } from 'react';

class StarRating extends Component {

     
    render() { 
        return ( 
            <div>
                <h3>Star Ratings</h3>
                <br />
                <br />
            </div>
         );
    }
}
 
export default StarRating;