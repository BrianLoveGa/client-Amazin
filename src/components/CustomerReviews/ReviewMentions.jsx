import React, { Component } from 'react';
import "./CustomerReviews.css" 


class ReviewMentions extends Component {


    render() { 

        return ( 
            <div>
                <div className="review-mention">
                    <h3>Read reviews that mention</h3>
                    <span><button className="button">filtered button</button></span>
                </div>
                   <span><button>top reviews</button></span>
                <br />
                <br />
                <div>
                    <div className="profile-avatar"><img className="profile-avatar-img" src="https://www.amazon.com/images/S/amazon-avatars-global/default._CR0,0,1024,1024_SX48_.png"></img>
                    <span className="profile-name">Customer Name</span></div>
                </div>

                <div >
                    <span>*Star rating icon* <a href="link to customer review summary page">Review title!</a></span>
                </div>
                    <span>Review Date</span>
                <div>
                    <span>Verified Purchase</span>
                </div>
                <div><p>Submitted review form goes here</p></div>
                <div className="row"><button>Helpful</button>
                    <a href="/">Comment</a>
                </div>
            </div>
         );
    }
}
 
export default ReviewMentions;