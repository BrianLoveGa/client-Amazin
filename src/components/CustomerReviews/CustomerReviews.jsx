import React, { Component } from "react";
import ReviewProduct from "./ReviewProduct";
import CustomerImages from "./CustomerImages";
import ReviewMentions from "./ReviewMentions";
import StarRating from "./StarRating";
import Grid from "@material-ui/core/Grid";
import CustomerReviewsDataService from "../../service/CustomerReviewsDataService";
import GoodChartMock from '../RatingSummary/goodChartmock';
import Ratings from "../productVariant/variant-Components/Ratings";

class CustomerReviews extends Component {
	constructor(props) {
		super(props);
		this.refreshCustomerReviews = this.refreshCustomerReviews.bind(this);
    }
    
    state={
        review: []
    }

	componentDidMount() {
		this.refreshCustomerReviews();
	}

    refreshCustomerReviews() {
        CustomerReviewsDataService.retrieveAllCustomerReviews()
        .then(
            response => {
                this.setState({ review: response.data })

            }
        )
    }

    mapping = () => {
        console.log(this.state.review)
        return(
            <div>
            {this.state.review.map( resp => <div><ReviewMentions id={resp.id} rate={resp.rating} body={resp.reviewBody} title={resp.reviewTitle} tag={resp.reviewTag} name={resp.reviewer}/></div>)}
            </div>
        )
    }
    
    render() { 
        return ( 
            <div>
                <h2> Customer Reviews</h2>
                
                <Grid container justify="flex-start" alignItems="flex-start" id="left-grid">
                    <Grid item sm={4}>
                        <div><GoodChartMock/></div>
                        
                        <div id="review-feature"> <hr/> <h3><strong>By feature</strong></h3><Ratings id={this.props.params} /></div>
                        <a>See more</a><hr/>
                          
                        <div className="review-product"><ReviewProduct /></div>
                    </Grid>
                    
                    <Grid item sm id="right-grid">
                        <div><CustomerImages /></div> 
                        <br />
                        {this.mapping()}
                       {/* {this.state.review.map( <div><ReviewMentions id={resp.id} rate={resp.rating} body={resp.reviewBody} title={resp.reviewTitle} tag={resp.reviewTag} name={resp.reviewer}/></div>)} */}
                    </Grid>
                </Grid>
            
            </div>
         );
    }
}

export default CustomerReviews;
