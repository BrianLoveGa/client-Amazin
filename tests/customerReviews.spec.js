import React from 'react';
import ReactDOM from 'react-dom';
import CustomerReviews from '../src/components/CustomerReviews/CustomerReviews';

import { render } from '@testing-library/react';

import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";


it("it renders", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CustomerReviews/>, div);
})

it("matches snapshot", () => {
    const tree = renderer.create(<CustomerReviews/>).toJSON();
    expect(tree).toMatchSnapshot();
})