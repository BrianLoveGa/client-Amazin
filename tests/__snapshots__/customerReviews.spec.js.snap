// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h1>
     Customer Reviews
  </h1>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-start"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4"
    >
      <div>
        <div>
          <h3>
            Star Ratings
          </h3>
          <br />
          <br />
        </div>
      </div>
      <br />
      <div>
        <div>
          <h3>
            Review this product
          </h3>
          <h5>
            Share your thoughts with other customers
          </h5>
          <a
            href="link to review form"
          >
            <button>
              Write a customer review
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-true"
    >
      <div>
        <div>
          <h3>
            Customer Images
          </h3>
          <div>
            <img
              src="http://placehold.it/150x150"
            />
          </div>
        </div>
      </div>
      <br />
      <div>
        <div>
          <div
            className="review-mention"
          >
            <h3>
              Read reviews that mention
            </h3>
            <span>
              <button
                className="button"
              >
                filtered button
              </button>
            </span>
          </div>
          <br />
          <br />
          <div
            className="dropdown"
          >
            <button
              className="dropbtn"
            >
              Top reviews
            </button>
            <div
              className="dropdown-content"
            >
              <a
                href="#"
              >
                Most recent
              </a>
            </div>
          </div>
          <br />
          <br />
          <div>
            <div
              className="profile-avatar"
            >
              <img
                className="profile-avatar-img"
                src="https://www.amazon.com/images/S/amazon-avatars-global/default._CR0,0,1024,1024_SX48_.png"
              />
              <span
                className="profile-name"
              >
                Customer Name
              </span>
            </div>
          </div>
          <tr>
            <td>
              <span>
                *Star rating icon* 
                <a
                  href="link to customer review summary page"
                >
                  Review title!
                </a>
              </span>
            </td>
          </tr>
          <div>
            Review Date
          </div>
          <div>
            Verified Purchase
          </div>
          <div>
            <p>
              Submitted review form goes here
            </p>
          </div>
          <table>
            <tr>
              <td>
                <button>
                  Helpful
                </button>
              </td>
              <td>
                <a
                  href="/"
                >
                  Comment
                </a>
              </td>
            </tr>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
